python_modules/dagster/dagster/_core/definitions/asset_key.py

```diff
@@ -101,7 +101,7 @@ def to_python_identifier(self, suffix: Optional[str] = None) -> str:
         if suffix is not None:
             path.append(suffix)
 
-        return "__".join(path).replace("-", "_")
+        return "__".join(path).replace("-", "_").replace(".", "_")
```

Comment:
I'm pretty sure we have a centralized place for doing this; perhaps we should be centralizing this as well (likely too late as it would be a breaking change.

At the very least, want to understand why this is necessary.

Reply:
we are using `AssetKey.to_python_identifier()` in a new place that has a wider variety of input asset keys, so this method was updated to be better. In reality, this method should be able to accept any asset key and convert it into a valid python identifier using some sort of regex (technically it's currently impossible to use `specs` as input for `@multi_asset` etc. if your asset key is sufficiently nasty), I was just lazy

Reply:
sounds good